### DISCUSSION TITLE
Announce dumi 2.3.0 🎉🎉🎉

### DISCUSSION THREAD

**Discussion by PeachScript** [umijs/dumi#2085] [General]
我们很高兴地宣布，dumi 2.3.0 版本已经正式发布！

由于这个版本包含的新特性都涉及相当大的改造工作量，太长的战线也使得我们需要花大量的时间在方案的磋商、验证、修改之中，再加上时间线拉长不可避免出现精力的分散，纵使我们从去年 8 月就启动了研发，还是拖到今天才正式发布，希望这些新的特性能够帮助你更好地进行组件库开发，构建组件库文档或静态站点。

### 1️⃣ 支持组件 demo 的在线实时编辑和预览

demo 的实时编辑预览是 dumi 1 开始就呼声很高的特性，但我们一直没找到轻量级的方案来实现它，倘若方案太过重量级，又远不如 CodeSandbox 等平台的体验来得更好。

不过这次我们终于找到了突破口，并实现了轻量级的实时编辑 + 预览：

<img src="https://mdn.alipayobjects.com/huamei_i7qs1c/afts/img/A*CX_gQYtqT8cAAAAAAAAAAAAADs-oAQ/original" width="640" />

它当然有一些限制，比如不支持引入新的依赖、乞丐版的编辑器、暂时不能修改非 entry 文件、不支持样式文件等等，但它足够方便快捷，与原本就有的 CodeSandbox 外链能力形成互补：
- 少量变更测试功能，使用 live demo 即改即看、即用即走
- 复杂变更验证功能，跳转 CodeSandbox/StackBlitz 使用完整编辑预览能力

### 2️⃣ 支持 Vue.js 技术栈

dumi 从 1.0 开始就支持自定义 demo 技术栈，2.0 开始还对技术栈开放能力做了全面升级，但 dumi 始终没有真正支持 React 以外的技术栈；而这次有了社区新同学的强势加入，在打磨、完善技术栈开放能力的同时，还基于这套能力完成了 Vue.js 技术栈的支持：

<img src="https://mdn.alipayobjects.com/huamei_i7qs1c/afts/img/A*1z3UTZ2kDb8AAAAAAAAAAAAADs-oAQ/original" width="640" />

不仅 Vue.js 的 demo 可以正常编译和渲染，甚至也支持 demo 的实时编辑和预览，以及 Vue.js 组件的 API 自动表格能力，这是一套功能完备的技术栈实现！

如果你的应用是其他技术栈，比如 Angular、Taro、小程序，也欢迎基于这套开放能力实现相应配套，帮助 dumi 完善生态。

### 3️⃣ 页面文本数据及 demo 数据的按需加载

dumi 2 有开箱即用的全文搜索功能，但在之前的版本中，dumi 会一次性加载所有的页面的文本数据和 demo 数据用于建立索引，这会导致大型站点的首屏性能变成一场灾难！比如在 Ant Design 官网中，入口 JS 文件的大小曾一度冲上 10MB，可想而知网站首屏的性能会受到怎样的影响。

而在 2.3.0 版本中，我们实现了页面数据的按需加载，只有当用户访问某个页面或进行搜索时，才会加载相应的数据，在减少入口 JS 文件尺寸的同时也降低了对网站首屏性能的影响。

<img src="https://mdn.alipayobjects.com/huamei_i7qs1c/afts/img/A*nwZxRb2u7LUAAAAAAAAAAAAADs-oAQ/original" width="640" />

除了上述特性外，我们这次还带来了 SourceCode 分组、默认 loading 动画等新特性，欢迎升级体验。

## 🔮 未来计划 🔮

虽然 2.3.0 刚刚发布，但 2.4.0 蓄力已有时日，以下是 2.4.0 可能会包含的新特性：

- 使用 UmiJS 团队倾力打造的 Rust 构建工具替代 Webpack，带来闪电般的本地研发体验
- 非实验性质的自动 API 表格，提供更快、更准确、定制化成都更高的自动 API 表格能力
- 全新的默认主题 UI 及交互体验
- 提供官方插件辅助组件开发者评估组件的无障碍可访问性（A11y）
- dumi 配套的 NPM 包研发工具 father 支持构建 Vue.js 组件库
- ……

## 🙌 感谢 🙌

dumi 2.3.0 最终得以发布，离不开众多 Collaborator 的努力，感谢你们让开源变得更加美好：

- @jeffwcx 投入大量心血实现 Vue.js 技术栈
- @Wxh16144 实现 SourceCode 分组及默认 loading 动画等特性
- @ZombieJ @MadCcc 结对实现路由数据按需加载特性
- @MadCcc 探索并验证 demo 实时编辑预览的方案
- @Ant-Design-Team 对 dumi 2.3.0 alpha 的踩坑及信任
- @所有持续在帮 dumi 完善的社区贡献者们

<img src="https://mdn.alipayobjects.com/huamei_i7qs1c/afts/img/A*HdjkTqPgcxsAAAAAAAAAAAAADs-oAQ/original"  width="640" />

也感谢所有支持 dumi、选择 dumi 的开源社区伙伴，让我们一起为更好的组件研发体验而努力！

**Comment by Wxh16144** [comment]
很期待 `2.4.0`自动 API 功能，社区太需要了！！

**Comment by HytonightYX** [comment]
期待一波 rust 构建工具，目前在其它项目中从 webpack 切换到 rust 构建工具都非常满意
